### PR TITLE
PHP 8.3 support added

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by Ocramius** [comment]
I went over a few of your PRs, @glo71317: as much as I appreciate that Adobe wants PHP 8.3 support, we need builds to be green, in order for a merge to happen 😬

These patches show intent, but are of no help, if they stay red: it's just more work on @laminas/technical-steering-committee

**Comment by glo71317** [comment]
> I went over a few of your PRs, @glo71317: as much as I appreciate that Adobe wants PHP 8.3 support, we need builds to be green, in order for a merge to happen 😬
> 
> These patches show intent, but are of no help, if they stay red: it's just more work on @laminas/technical-steering-committee

@Ocramius completely agree :) But there is dependencies on other packages so builds are red, In order to make the build green following green build need to merge
https://github.com/laminas/laminas-captcha/pull/31
https://github.com/laminas/laminas-code/pull/194
https://github.com/laminas/laminas-loader/pull/12
https://github.com/laminas/laminas-uri/pull/21
https://github.com/laminas/laminas-json/pull/18
https://github.com/laminas/laminas-math/pull/19
https://github.com/laminas/laminas-recaptcha/pull/33
https://github.com/laminas/laminas-soap/pull/42
https://github.com/laminas/laminas-text/pull/62
https://github.com/laminas/laminas-cache-storage-adapter-memory/pull/53

If you have any pr comments then feel free leave the comments we will resolve the pr comments asap else proceed for merge above prs to speed up to make green other prs :)

**Comment by Ocramius** [comment]
Yes, but the current effect is that I have a huge load of emails, and I'm just playing whack-a-mole with them.

That's why I'm saying it isn't helpful 😁

I will go through the list from your comment though 👍

**Comment by Ocramius** [comment]
Meanwhile, all mentioned patches should have associated releases 👍

**Comment by boesing** [comment]
For those packages where transitive dependencies are not yet supporting PHP 8.3 and therefore the checks are not passing, you can add `ignore_php_platform_requirements` to `.laminas-ci.json` in project root.

```json
{
     "ignore_php_platform_requirements": {
        "8.3": true
     }
}
```

If checks are still failing afterwards, that might be related to transitive dependencies not supporting PHP 8.3.
In case these are laminas components, this might be raised in those components.

In any case, we have to wait until there are patched versions for those components which lead to failing tests due to the PHP update.

**Comment by glo71317** [comment]
@Ocramius I have updated all associated release for patches and tests are green now.
Can you review and merge this pr if there is no concern?

**Comment by glo71317** [comment]
@Ocramius @gsteel PR is ready more than a week, No one PR is reviewing?

File: psalm-baseline.xml [hunk 1]
```diff
@@ -363,6 +372,9 @@
       <code>onExtract</code>
       <code>onHydrate</code>
     </InternalMethod>
+    <DeprecatedMethod occurrences="5">
+      <code>$this->returnValue</code>
+    </DeprecatedMethod>
```
Comment (gsteel) [inline psalm-baseline.xml:377]: These deprecations can be fixed in the test with `MockObject::willReturn($whatever)`

File: psalm-baseline.xml [hunk 2]
```diff
@@ -519,6 +531,19 @@
...
+  </file>
+  <file src="test/HydratorTestTrait.php">
+    <DeprecatedMethod occurrences="4">
+      <code>$this->returnValue</code>
+    </DeprecatedMethod>
+  </file>
+  <file src="test/NamingStrategy/CompositeNamingStrategyTest.php">
+    <DeprecatedMethod occurrences="4">
+      <code>$this->returnValue</code>
+    </DeprecatedMethod>
```
Comment (gsteel) [inline psalm-baseline.xml:546]: These deprecations can be fixed in the test with `MockObject::willReturn($whatever)`

File: psalm-baseline.xml [hunk 3]
```diff
@@ -172,6 +172,9 @@
       <code>$value</code>
       <code>$value</code>
     </MixedArgument>
+    <ArgumentTypeCoercion>
+      <code>$pattern</code>
+    </ArgumentTypeCoercion>
```
Comment (gsteel) [inline psalm-baseline.xml:177]: This can be solved by asserting that `preg_match` receives a `non-empty-string`

File: psalm-baseline.xml [hunk 4]
```diff
@@ -208,6 +211,9 @@
       <code>$matches[2]</code>
       <code>$value[0]</code>
     </MixedArrayAccess>
+    <ArgumentTypeCoercion>
+      <code>$pcreInfo->pattern</code>
+    </ArgumentTypeCoercion>
```
Comment (gsteel) [inline psalm-baseline.xml:216]: This can be solved by asserting that `preg_match` receives a `non-empty-string`

**Review by gsteel — CHANGES_REQUESTED** [review]
Thanks @glo71317 - There are a few easily solved but baselined issues here that have been handled in #118 - so I'll prefer that patch over this one.

**Comment by gsteel** [comment]
Handled in #118

**Pull request closed without merging**